### ISSUE TITLE
[Request]: Use Real Program Icon for the Program Window?

### ISSUE THREAD

**Issue by Perkolator** [bottlesdevs/Bottles#2985]
### Tell us the problem or your need

The running program window in the taskbar always show the Bottles icon, not the actual program icon.

![bott](https://github.com/bottlesdevs/Bottles/assets/53913724/66b9f5e6-0e7a-4ed5-8472-862cbb5abbc4)

(Screenshot is from Linux Mint 20.3 Cinnamon & X11)

### Describe the solution you'd like

Would it be possible to use the actual program icons for the windows?

**Comment by PacBilly** [comment]
Yeah, I second this request. If you use the non-Flatpak version of Bottles, it handles this properly and shows program icon, but with the Flatpak it shows the Bottles icon instead. Drives me crazy, but the flatpak version is less buggy in other regards, so it's the lesser of two evils, I guess. But even in the Bottles team's example .desktop entry YouTube video, you can see it's running the wrong icon. It needs to appear is though the app is running natively, as it does in the non-Flatpak version.

**Comment by M0JXD** [comment]
I would love this as well, especially given that the guide/docs heavily encourage the use of the Flatpak.

In the meantime, is there any way to set panel/taskbar icons manually in the Flatpak version? I now have multiple apps using Bottles and running them at the same time gets very confusing very quickly!

**Comment by Ambient-Impact** [comment]
I'd love to see this as well, but it looks like this a security restriction of Flatpak on Wayland; see #flatpak/flatpak#4751, so not sure what Bottles can do here.

**Comment by ParticleMon** [comment]
In the OP, the DE is Cinnamon, which is likely on X11 since Cinnamon's Wayland support is still in Alpha.

But if this Flatpak security restriction is "working as designed, although the design is not ideal" (https://github.com/flatpak/flatpak/issues/4751#issuecomment-1041520941) and it also applies to X11, then might the providing of an AppImage be expedited, or at least a .deb?

**Comment by jntesteves** [comment]
I think this issue might get solved when desktop compositors implement this new (staging) Wayland protocol and Flatpak (and maybe Wine also) starts using it: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/269

**Comment by Ambient-Impact** [comment]
Interestingly, I switched to Nobara 40 which runs KDE Plasma 6 on Wayland, and apps launched from Bottles use their own icons rather than Bottles' icon, so I'm wondering if Plasma has already implemented `xdg-toplevel-icon` or has their own workarounds for this?

**Comment by ogwhic** [comment]
This problem doesn't occur in KDE or Cinnamon.

I think it's an issue with GNOME and other DEs when running under Wayland but I've no idea where to report it or what the actual underlying issue is.

Tested on Fedora 41 (GNOME, KDE and Cinnamon).

I do agree it's very annoying.

**Comment by Perkolator** [comment]
> This problem doesn't occur in KDE or Cinnamon.

The screenshot in the 1st post is from Linux Mint Cinnamon (and X11).

**Comment by M0JXD** [comment]
The issue still occurs for me on Linux Mint 22 Cinnamon (X11).

**Comment by aleex5** [comment]
> Interestingly, I switched to Nobara 40 which runs KDE Plasma 6 on Wayland, and apps launched from Bottles use their own icons rather than Bottles' icon, so I'm wondering if Plasma has already implemented `xdg-toplevel-icon` or has their own workarounds for this?

Check if it is a flatpak version or a packaged version of fedora, because in fedora there is a version that is not flatpak

**Comment by jrgn9** [comment]
I found a solution that worked for me on Cinnamon. I am running Mint 22.3 with Bottles installed as Flatpak. I did this for Qobuz, but might work for other programs too. 

I edited the .desktop file created by Bottles in `~/.local/share/applications`. First I added `StartupWMClass=Qobuz.exe` under [Desktop Entry]. I already had `StartupWMClass=Qobuz`, so I just added the .exe suffix. Then I renamed the whole .desktop file to `Qobuz.exe.desktop`. Then the icon showed up, both in panel and in main menu. 

When I changed the .desktop file name, the name of the program also changed to Qobuz.exe.desktop, so I had to go back and edit Name in the .desktop entry. `#!/usr/bin/env xdg-open` was also added to the top of the file. I tried both using it and removing it, but it seems like it made no difference.
